src/components/profile-update/profile-update.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { Form, Button, Card, CardGroup, Container, Col, Row } from 'react-bootstrap';
+import { BrowserRouter as Router, Link } from 'react-router-dom';
+
+import React, { useState } from 'react';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+
+
+
+export function ProfileUpdate(props) {
+
+
+    console.log("props: ", props);
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [email, setEmail] = useState('');
+    const [birthday, setBirthday] = useState('');
+
+    const handleUpdateUser = (e) => {
+        e.preventDefault();
+        /* Send a request to the server for authentication */
+        axios.put('https://morning-badlands-52426.herokuapp.com/users', {
+            Username: username,
+            Password: password,
+            Email: email,
+            Birthday: birthday
+        })
+            /* then call props.onRegistration(username) */
+            .then(response => {
+                const data = response.data;
+                console.log(data);
+                alert("user updated");
+                // props.onRegistration(data);
+                props.history.push("/");
+            })
+            .catch(e => {
+                console.log('wrong format or incomplete data');
+                alert('fill in all the fields in correct format');
+            });
+    };
+
+
+
+    return (
+        <Router>
+            <Container>
+                <div className="form-container">
+                    <Row>
+                        <Col>
+                            <CardGroup>
+                                <Card>
+                                    <Card.Body>
+                                        <Card.Title>Update User</Card.Title>
+                                        <Form>
+                                            <Form.Group>
+                                                <Form.Label>Username:Profile Name:{user.Username}</Form.Label>
```

Comment:
This part of the application is breaking. Don't you think you are trying to access `user` yet it is not defined anywhere above? Here is the error in my console

<img width="1440" alt="Screenshot 2021-11-22 at 09 11 19" src="https://user-images.githubusercontent.com/25340897/142814264-23478f0c-2232-4366-ab4a-a9c439443b45.png">

Reply:
That's the thing I don't know how to resolve.
I would like the const hadleUpdateUser send the PUT request with current User Name at the end of URL:
https://morning-badlands-52426.herokuapp.com/users/CURRENT_LOGGED_IN_USER_ NAME

My logic is that somewhere in the App we have the details of the user Name just don't know how to source it.
Then we should be ok for the Auth Token as we are still the current user (_id stays the same). 
What do you think? HELP :)

Reply:
You can get the user from the props. I believe you are passing the user into the component in the main-view render function. But in here, you are not getting the user from the props. That may be why. Feel free to spread the props as well.